app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+  private
+
+    def require_admin
+      render file: 'public/404' unless current_admin
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.